packages/apollos-ui-analytics/src/Provider.js

```diff
@@ -1,168 +1,173 @@
-        createResolvers({
-          trackFunctions,
-          identifyFunctions,
-          useServerAnalytics,
-        })
-      );
-      return (
-        <AnalyticsContext.Provider
-          value={{
-            track: createTrack({ client }),
-            identify: createIdentify({ client }),
-            notify: (error) => notifyFunctions.forEach((fn) => fn(error)),
-          }}
-        >
-          {children}
-        </AnalyticsContext.Provider>
-      );
-    }}
-  </ApolloConsumer>
-);
+  church = Config.CHURCH_HEADER,
+}) => {
+  const track = useCoreTrack();
+  const identify = useCoreIdentify();
+
+  const clientsWithDefaults = useMemo(() => [{ track, identify }, ...clients], [
+    clients,
+    track,
+    identify,
+  ]);
+
+  const client = useMemo(() => {
+    return {
+      screen: (...args) =>
+        Promise.all(
+          clientsWithDefaults?.map((c) => {
+            // backwards compatibility: if no screen method is provided, use track
+            // This is mainly used to support Legacy Amplitude + CoreAnalytics
+            // which does not provide a "screen" method.
+            if (!c?.screen && c.track) {
+              return c.track(...args);
+            }
+            return c?.screen && c.screen(...args);
+          })
+        ),
+      track: (eventName, properties, ...args) => {
+        // backwards compatibility, older method syntax was to pass an object
+        if (typeof eventName === 'object') {
+          properties = eventName.properties;
+          eventName = eventName.eventName;
+        }
 
-export const useTrack = () => {
-  const ctx = useContext(AnalyticsContext);
-  return ctx.track;
+        return Promise.all(
+          clientsWithDefaults?.map(
+            (c) => c?.track && c.track(eventName, properties, ...args)
+          )
+        );
+      },
+      identify: (...args) =>
+        Promise.all(
+          clientsWithDefaults?.map((c) => c?.identify && c.identify(...args))
+        ),
+      flush: (...args) =>
+        Promise.all(
+          clientsWithDefaults?.map((c) => c?.flush && c.flush(...args))
+        ),
+      group: (...args) =>
+        Promise.all(
+          clientsWithDefaults?.map((c) => c?.group && c.group(...args))
+        ),
+      alias: (...args) =>
+        Promise.all(
+          clientsWithDefaults?.map((c) => c?.alias && c.alias(...args))
+        ),
+      reset: (...args) =>
+        Promise.all(
+          clientsWithDefaults?.map((c) => c?.reset && c.reset(...args))
+        ),
+    };
+  }, [clientsWithDefaults]);
+
+  // Auto track user
+  const { data } = useQuery(GET_ANALYTICS_USER, {
+    returnPartialData: true,
+    errorPolicy: 'ignore',
+  });
+  useEffect(() => {
+    if (!data?.currentUser?.profile?.id) {
+      return;
+    }
+    client.identify(data.currentUser.profile.id, {
+      ...data.currentUser.profile,
+      avatar: data.currentUser.profile.photo?.uri,
+    });
+  }, [client, data?.currentUser?.profile]);
```

Comment:
Removing `client` here stops an infinite identify loop from happening. Might be better options? 
```suggestion
  }, [data?.currentUser?.profile]);
```